src/state/wifi_collection.cpp

```diff
@@ -0,0 +1,102 @@
+#include "wifi_collection.h"
+#include "qjsonarray.h"
+
+
+namespace Valeronoi::state {
+
+wifi_collection::wifi_collection()
+{
+    clear();
+}
+
+void wifi_collection::clear()
+{
+    m_known_wifis.clear();
+    m_known_wifi_ids.clear();
+}
+
+int wifi_collection::get_or_create_wifi_id(Valeronoi::robot::WifiInformation wifi_info)
+{
+    int nRet = get_wifi_id(wifi_info.bssid());
+
+    if (nRet < 0) {
+        nRet = add_wifi(wifi_info);
+    }
+
+    return nRet;
+}
+
+int wifi_collection::add_wifi(Valeronoi::robot::WifiInformation wifi_info)
+{
+    m_known_wifis.append(wifi_info);
+    m_known_wifi_ids.append(m_known_wifis.size()-1);
+
+    if (m_known_wifis.size() != m_known_wifi_ids.size())
+    {
+        // out of sync.
+        throw "Vector size missmatch! this is bad";
+    }
+
+    emit signal_wifi_list_updated();
+    emit signal_new_wifi_added(wifi_info);
+
+    return m_known_wifi_ids.last();
+}
+
+QVector<Valeronoi::robot::WifiInformation> wifi_collection::get_known_wifis() const
+{
+    return m_known_wifis;
+}
+
+QJsonArray wifi_collection::get_json() const
+{
+    QJsonArray jArr;
+
+    for (int index = 0; m_known_wifis.size() > index ; ++index) {
+        QJsonObject wifi_access_point = m_known_wifis.at(index).get_json();
+        wifi_access_point.insert("index",m_known_wifi_ids.at(index));
```

Comment:
Why do we need the "index" attribute at all? A vector and a QJsonArray are already sorted (SSOT).

Reply:
the attribute is the mapped wifi Id. badly named here.. will change it to wifi_id

Reply:
because the id is not part of the Wifi Info. It´s appended in the collection, to make a lightwight wifi mapping.